Document the web dashboard feature (#1484)
grafana/k6-docs: docs/sources/next/results-output/web-dashboard/_index.md

---
title: Web Dashboard
excerpt: With the web-dashboard, you can track test results in real-time, and generate HTML test reports directly from your web browser.
weight: 200
---

# Web dashboard

k6 provides a built-in web dashboard running on the user's machine and enabling them to monitor their test results in real-time.

## Real-time test results

![Web dashboard screenshot](/media/docs/k6-oss/web-dashboard-overview.png)

> The dashboard provides a real-time overview of the performance observed by k6 while a test is running, and helps to identify potential reliability issues as they occur.

Author: Sounds good 👍🏻

## HTML test reports

The dashboard enables to generate detailed, downloadable HTML reports directly from the dashboard. These reports are self-contained, making them ideal for sharing with your team.

![HTML test report screenshot](/media/docs/k6-oss/web-dashboard-report.png)

> The HTML test report can be accessed from the dashboard's menu, by clicking on the "Report" button.

Reviewer: ```suggestion
You can access the HTML test report by clicking the "Report" button on the dashboard's menu.
```

![HTML test report generation button](/media/docs/k6-oss/web-dashboard-report-button.png)

> 

Reviewer 2: Perhaps a sentence could be written about how the report can be automatically saved using the `K6_WEB_DASHBOARD_EXPORT` environment variable, without starting a browser.

> ## How to use

Reviewer: I think we should move this section above the "HTML test reports" one, so it goes: Introduction, How to use, HTML reports, Dashboard options.

    /  \/    \    |     (   /   ‾‾\
   /          \   |  |\  \ |  (‾)  |
  / __________ \  |__| \__\ \_____/ .io

     execution: local
        script: ../extensions/xk6-dashboard/script.js
 web dashboard: http://127.0.0.1:5665
        output: -
```

By default, when active, the web dashboard is exposed on localhost port `5665`. You can however configure.

## Dashboard options

The web dashboard can be configured using environment variables:

| Environment variable               | Description                                   | Default value |
| ---------------------------------- | --------------------------------------------- | ------------- |
| `K6_WEB_DASHBOARD`                 | Enable the web dashboard                      | `false`       |
| `K6_WEB_DASHBOARD_HOST`            | Host to bind the web dashboard to             | `localhost`   |
| `K6_WEB_DASHBOARD_PORT`            | Port to bind the web dashboard to             | `5665`        |
| `K6_WEB_DASHBOARD_PERIOD`          | Period in seconds to update the web dashboard | `10s`         |
| `K6_WEB_DASHBOARD_OPEN`            | Open the web dashboard in the default browser | `false`       |

> | `K6_WEB_DASHBOARD_REPORT_FILENAME` | Filename to export the HTML test report to    | `report.html` |

Reviewer 2: `K6_WEB_DASHBOARD_EXPORT` is the exported report file name env var (not `K6_WEB_DASHBOARD_REPORT_FILENAME`)

---
title: Web dashboard
description: Track test results in real-time with the web-dashboard and generate HTML test reports directly from your web browser.
weight: 200
---

# Web dashboard

k6 provides a built-in web dashboard that you can enable to visualize and monitor your tests results in real-time.

![Web dashboard screenshot](/media/docs/k6-oss/web-dashboard-overview.png)

The dashboard provides a real-time overview of the performance observed by k6 while a test is running, and can help you identify potential reliability issues as they occur.

## Generate HTML test reports

> You can generate detailed, downloadable HTML reports directly from the dashboard. These reports are self-contained, making them ideal for sharing with your team.
> 
> ![HTML test report screenshot](/media/docs/k6-oss/web-dashboard-report.png)
> 
> You can access the HTML test report by clicking the "Report" button on the dashboard's menu.
> 
> ![HTML test report generation button](/media/docs/k6-oss/web-dashboard-report-button.png)

Reviewer: ````suggestion
You can generate detailed, downloadable HTML reports directly from the web dashboard or the command line. These reports are self-contained, making them ideal for sharing with your team.

![HTML test report screenshot](/media/docs/k6-oss/web-dashboard-report.png)

### Generate report from web dashboard

To generate a report from the web dashboard, click **Report** on the dashboard's menu.

![HTML test report generation button](/media/docs/k6-oss/web-dashboard-report-button.png)

### Generate report from the command line

To automatically generate a report from the command line once the test finishes running, use the `K6_WEB_DASHBOARD_EXPORT` option. For example:

```shell
K6_WEB_DASHBOARD=true K6_WEB_DASHBOARD_EXPORT=html-report.html k6 run script.js
```
````

@oleiade just a suggestion here taking into account @szkiba feedback to mention `K6_WEB_DASHBOARD_EXPORT`.
